### ISSUE TITLE
tracking code incorrectly treated

### ISSUE THREAD

**Issue by barnoldbert** [2Abendsegler/GClh#2507]
### Describe the bug

the search field in page header does nor resolve some trackable codes correctly.

### To Reproduce

go to anay page where GClh is working
enter in the searchfield in the top BMYTK6
it the tries to open https://www.geocaching.com/plan/lists/BMytk6  that does not exist

### Expected behavior

instead it should try to open 
https://www.geocaching.com/plan/lists/BMytk6

### OS

Windows

### Browser

Firefox, Chrome

### GClh Version

0.15.1

### Additional context

I tried several other tracking codes, they are treated correctly

**Comment by 2Abendsegler** [comment]
We could still build TL.

Instead of a dialogue, we could work with an identification in the case of a tracking number/tracking code. So for our example `tn bmytk6` and `tc bmytk6`. If no identification is specified, everything stays as usual. Tracking numbers/tracking codes can also be specified as before, as long as there is no overlap with the other rules. If one of these identifications is specified, then it is a tracking number/tracking code.

Maybe we should finally provide a description of the possible inputs, perhaps similar to the auxiliary fields in the config, i.e. with a question mark, or perhaps even better, show title (multi-line) by hovering over the field.

**Comment by capoaira** [comment]
I like the idea. But `tn` and `tc` is in my opinion not clear enough, 'cause one of this means the tracking code of the TB and the other means caches, TBs, profiles, BMs (not just TBs). Did we need two identifications or is one, to force tracking code, enough?
The RegEx should also be updated to accept whitespace:
```js
search.match(/^\s*(FT\s+)?((?:GC|TB|GT|PR|BM|GL|TL)[A-Z0-9]+)\s*$/i)
```

**Comment by 2Abendsegler** [comment]
The official name for this code according to TB Listing is “Tracking Number”. In this respect, an abbreviation `tn` would make sense. It doesn't have to be two.

**Comment by capoaira** [comment]
I understand both, and I'm not sure which one I prefer. The thing with the prefix is, that it might not be DAU compatible. A dialog is clear to all, but of course it is an extra step for all how enter a 6-characters code. And I think that the most users use GC or TB numbers for Caches and Trackables. Bookmarks, Profiles and Logs might more shared as link than just the code.

The tracking codes have a prefix of two characters, which is related to the type (For example CV for Community Volunteer tags, or GS for Lackey tags) followed by 4 random characters. So we have 33⁴ = 1.185.921 possibilities for each prefix. But there are multiple types that can use one prefix in common. So it might not be assigned all possible codes (33⁶ = 1.291.467.969).
Let's say 100 of the 1089 possible prefixes will be used, there are 100 * 33⁴ = 118.592.100, which will be assigned. (My 45 coins have 25 different prefixes, which indicates that more than 100 prefixes are in use). That should be enough that we don't have to think about 7-character codes.



> the condition could be: if the code has 6 characters AND starts with one of the standard combinations GC|TB|GT|PR|BM|GL|TL...
> then ask if the user wants to find a travelbug.

That is what I mean. 



> BTW I only use the search field to find a cache or a TB. In general I never have the code of one of the other types of items at hand.

Yes, I too, but also for GeoTours. But that does not mean that other not use one of the others.



> The official name for this code according to TB Listing is “Tracking Number”. In this respect, an abbreviation `tn` would make sense. It doesn't have to be two.

Oh. I'm stupid. I thought, that you mean one for Tracking number and one for GC, TB, ... codes 🤦🏻‍♂️. Forgot what I said, we can of course support more abbreviation for that 😅

**Comment by 2Abendsegler** [comment]
> The thing with the prefix is, that it might not be DAU compatible.

Yes, but this whole field is not DAU compatible. I don't think many users know what you can do with the field.

A `tn` abbreviation does not always have to be used, only when there is an overlap, as here with `bm`. 

> That should be enough that we don't have to think about 7-character codes.

Unless gaps are intentionally left to prevent to discover TBs with arbitrary codes.

I think the effort to develop the dialogue is not justified, when there is such a tiny and simple intervention, that can achieve the same thing. The effort you have to put into this would be much more beneficial elsewhere.

Requirements for a new dialogue. What comes to mind shortly:
- The new dialog should be usable as an alternative to the current feature, new config parameter (false).
- For a dialog, the design of Config, Sync, User Search should be used.
- The new dialogue should be integrated into their color scheme.
- Config and Sync must be selectable when the new dialog is active and must not be disturbed.
- The new dialog has to be tested on all geocaching sites, because numerous different styles are now used on the geocaching sites.

---  
Off topic:
> My 45 coins

Are they all in your hands? Are they all displayed in the log form?

**Comment by capoaira** [comment]
> Yes, but this whole field is not DAU compatible. I don't think many users know what you can do with the field.

Really? I thought a search field is self explaining.

> I think the effort to develop the dialogue is not justified, when there is such a tiny and simple intervention, that can achieve the same thing. The effort you have to put into this would be much more beneficial elsewhere.

Yes, that is right. Maybe we can use the find player as template, so that we can ensure, that the dialog works on all pages.


---
> Off topic:
> 
> > My 45 coins
> 
> Are they all in your hands? Are they all displayed in the log form?

Not directly. I moved all into my collection. So these will not be shown on the log page

**Comment by 2Abendsegler** [comment]
> > Yes, but this whole field is not DAU compatible. I don't think many users know what you can do with the field.
> 
> Really? I thought a search field is self explaining.

😂 Yes, that's right. But it doesn't even say that it is a search field. And that's not documented anywhere, I don't think. And I didn't know that you could also search for logs from caches, but not logs from TBs.

> > I think the effort to develop the dialogue is not justified, when there is such a tiny and simple intervention, that can achieve the same thing. The effort you have to put into this would be much more beneficial elsewhere.
> 
> Yes, that is right. Maybe we can use the find player as template, so that we can ensure, that the dialog works on all pages.

You will probably need additional elements, such as radio buttons, which are not used on the find player template. But yes, this could be a start.

For example, with #2421 or #1473, many users would be much happier. ;)

**Comment by 2Abendsegler** [comment]
There was a new issue on this topic:
[[Navi Search] Searching for trackables does not work for trackables starting with "GT"](https://github.com/2Abendsegler/GClh/issues/2654)

**Comment by 2Abendsegler** [comment]
We have implemented a solution with #2834:

With the approach the behaviour of the navi search stays the same, but if you use ctrl+enter, you can force, searching a tracking code.
If the code is not 6 digits, it falls to the default behavior and looks for GC, TB, etc codes.
